src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -2084,8 +2057,31 @@ private Pal.journal_entry PrepareToWriteToJournal(LowLevelTransaction tx, ref Pa
             var overhead = sizeOfPagesHeader + (long)numberOfPages * sizeof(long);
             var overheadInPages = checked((int)(overhead / Constants.Storage.PageSize + (overhead % Constants.Storage.PageSize == 0 ? 0 : 1)));
 
+            var freedPages = tx.GetFreedPages();
+
+            if (freedPages is { Count: > 0 })
+            {
+                // we need to remove the freed pages that were reused and modified later on during the transaction
+                
+                foreach (var page in txPages)
+                {
+                    if (freedPages.Remove(page.PageNumberInDataFile) && page.NumberOfPages > 1)
```

Comment:
Invert this, we want to use the short-circuit to avoid the HashSet operation when skipping the loop completely is possible via `page.NumberOfPages`